src/main/java/com/management/HumanResources/service/ParseService.java

```diff
@@ -36,10 +39,60 @@ public Employee jsonToEmployee(JSONObject obj) {
         emp.setDept(obj.getString("dept"));
         emp.setManagerID(Long.parseLong(obj.getString("managerID")));
         emp.setId(Long.parseLong(obj.getString("id")));
-        //emp.setAddress(address); TODO
+        emp.setAddress(jsonToAddress(obj.getJSONObject("address")));
         return emp;
     }
 
-    //TODO: Add parsing for TimeOff list
+    public Address jsonToAddress(JSONObject obj) {
+        Address a = new Address();
+        a.setStreet(obj.getString("street"));
+        a.setCity(obj.getString("city"));
+        a.setState(obj.getString("state"));
+        a.setZipcode(obj.getString("zipcode"));
+        return a;
+    }
 
+	public List<EmployeeTime> toEmployeeTimesList(String json) {
+		JSONObject jsonList = new JSONObject(json);
+        List<EmployeeTime> employeeTimes = new ArrayList<EmployeeTime>();
+        jsonList.keySet().forEach(key -> 
+            employeeTimes.add(jsonToEmployeeTime(jsonList.getJSONObject(key))));
+        return employeeTimes;
+    }
+    
+    public EmployeeTime jsonToEmployeeTime(JSONObject obj) {
+        EmployeeTime et = new EmployeeTime();
+        et.setEmployeeId(obj.getLong("employeeId"));
+        et.setTotalHours(obj.getInt("totalHours"));
+        et.setHoursRemaining(obj.getInt("hoursRemaining"));
+        et.setAvailability(toEmployeeAvailabilityArray(obj.getString("availability")));
+        et.setTimeOffs(toEmployeeTimeOffsList(obj.getString("timeOffs")));
+        return et;
+    }
+
+    public String[] toEmployeeAvailabilityArray(String json) {
+		JSONObject jsonList = new JSONObject(json);
+        List<String> availabilityDays = new ArrayList<String>();
+        jsonList.keySet().forEach(key -> 
+            availabilityDays.add(jsonList.getString(key)));
+        String[] availabilityArray = new String[availabilityDays.size()];
+        return availabilityDays.toArray(availabilityArray);
```

Comment:
This might be getting changed since we converted to CSV. But if I'm not mistaken wouldn't L78 and L79 return an empty array?

Reply:
No, it copies the list to the array. But this is getting removed now in favor of CSV.

Reply:
Nice thank you!